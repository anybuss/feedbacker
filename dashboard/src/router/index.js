import { createRouter, createWebHistory } from "vue-router";

const Home = () => import("../views/Home/index.vue");
const Feedbacks = () => import("../views/Feedbacks/index.vue");

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/",
    name: "Feedbacks",
    component: Feedbacks,
  },
  {
    path: "/:pathMatch(.*)*",
    redirect: { name: "Home" },
  },
];

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

export default router;
