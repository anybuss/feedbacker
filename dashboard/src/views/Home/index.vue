<template>
  <custom-header
    @create-account="handleCreateAccount"
    @login="handleLogin"
  ></custom-header>
  <contact></contact>
</template>

<script setup>
import { onMounted } from "vue";
import { useRouter } from "vue-router";
import useModal from "@/hooks/useModal";
import CustomHeader from "./components/CustomHeader.vue";
import Contact from "./components/Contact.vue";

const router = useRouter();
const modal = useModal();

onMounted(() => {
  const token = window.localStorage.getItem("token");
  if (token) {
    router.push({ name: "Feedbacks" });
  }
});

function handleLogin() {
  modal.open({
    component: "ModalLogin",
  });
}

function handleCreateAccount() {
  modal.open({
    component: "ModalCreateAccount",
  });
}
</script>
