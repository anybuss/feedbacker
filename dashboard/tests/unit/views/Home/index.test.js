import Home from "@/views/Home/index.vue";
import { routes } from "@/router/index";
import { createRouter, createWebHistory } from "vue-router";
import { shallowMount } from "@vue/test-utils";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

describe("Home component", () => {
  it("should render home correctly", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
