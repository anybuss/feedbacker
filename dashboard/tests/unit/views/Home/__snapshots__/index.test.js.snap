// Vitest Snapshot v1

exports[`Home component > should render home correctly 1`] = `
"<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>"
`;

exports[`Home page > should render home correctly 1`] = `
"<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>"
`;
